service/history/chasm/components/activity/activity.go

```diff
@@ -0,0 +1,168 @@
+package activity
+
+import (
+	"time"
+
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/server/api/matchingservice/v1"
+	persistencepb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/service/history/chasm"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	Activity struct {
+		// ALL FIELDS MUST BE EXPORTED for reflection to work
+
+		// In V1, we will only support only one non-chasm.XXX field in the struct.
+		// and that field must be a proto.Message.
+		// TODO: define a serializer/deserializer interface?
+		//
+		// Framework will try to recognize the type and do serialization/deserialization
+		// proto.Message is recommended so the component get compatibility if state definition changes
+		State persistencepb.ActivityInfo // proto.Message
+
+		// At the end of a transition, framework will use reflection to understant the component
+		// tree structure.
+
+		// TODO: also support field name tag, so the fields can be renamed
+		Input  *chasm.Field[*common.Payload] `chasm:"lazy"`
+		Output *chasm.Field[*common.Payload] `chasm:"lazy"`
+
+		EventNotifier *chasm.Field[EventNotifier]
```

Comment:
I know this is just to show the pointer capabilities, but the real activity implementation will need to do more than just notify its parent, it'll need to be injected with a way to read the inputs and outputs. They won't be embedded payloads in the workflow case.

Reply:
I also had thought we wanted a way for the parent to be able to decide what notifications it cared about from its children, rather than the children needing to remember to notify the parent. I think that can still be implemented in terms of this a pointer just fine though.

Something as simple as the following I think would be nice, and isn't really even part of the framework, just a useful pattern. I think it keeps things a little more obvious for the activity implementer.

```go
func NewScheduledActivity(
	chasmContext chasm.MutableContext,
	params *NewActivityRequest,
	onComplete func(ActivityCompletedEvent)
) { ... }

// Then workflow constructs like
NewScheduledActivity(ctx, params, func(ce) { me.AsComponentPointer(ctx).OnCompletion(ce) })
```